go/libraries/doltcore/rowconv/row_converter.go

```diff
@@ -40,7 +40,7 @@ func newIdentityConverter(mapping *FieldMapping) *RowConverter {
 }
 
 // NewRowConverter creates a a row converter from a given FieldMapping.
-func NewRowConverter(mapping *FieldMapping) (*RowConverter, error) {
+func NewRowConverter(mapping *FieldMapping, specialBoolHandling bool) (*RowConverter, error) {
```

Comment:
This should just always be true, and it should only go one way: when we are importing the string TRUE or FALSE (case insensitive) into an integer column, auto convert to 1 or 0.

Or is there some compelling reason not to always do this?

Reply:
So `NewRowConverter` is used in 3 places from what I can see: imports, diffs, and conflicts. I haven't dove deep into exactly how the function is implemented in diffs and conflicts, so I can't say anything conclusive there.

I disagree that `"true"` and `"false"` strings should always autoconvert to 1 and 0 respectively. I prefer that all behavior is similar program-wide. I'm not too big of a fan of this change to begin with, as now SQL and imports deviate in behavior, and I'm **very** hesitant to apply it elsewhere.

Reply:
@bheni should weigh in here, since he is most familiar with this code's usage.

Agree with the idea we should be consistent. If this bothers you, one way to fix would be to pre-process the CSV / JSON files with a pipeline stage that specifically handles boolean values represented as strings. Then by the time it got here, you would already have numerical values.

As a matter of interface design, rather than passing a bool param to this method, just create a new method with a name that reflects its usage and the semantics you want it to have. Something like NewStringToBoolRowConverter or similar.